src/bms/command/channel.rs

```diff
@@ -175,6 +174,85 @@ pub enum PlayerSide {
     Player2,
 }
 
+/// A channel ID used in BMS files to identify channel types.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct ChannelId([u8; 2]);
+
+impl std::fmt::Debug for ChannelId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("ChannelId")
+            .field(&format!("{}{}", self.0[0] as char, self.0[1] as char))
+            .finish()
+    }
+}
+
+impl std::fmt::Display for ChannelId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}{}", self.0[0] as char, self.0[1] as char)
+    }
+}
+
+impl TryFrom<[char; 2]> for ChannelId {
+    type Error = [char; 2];
+    fn try_from(value: [char; 2]) -> core::result::Result<Self, Self::Error> {
+        Ok(Self([
+            char_to_base62(value[0]).ok_or(value)?,
+            char_to_base62(value[1]).ok_or(value)?,
+        ]))
+    }
+}
+
+impl TryFrom<[u8; 2]> for ChannelId {
+    type Error = [u8; 2];
+    fn try_from(value: [u8; 2]) -> core::result::Result<Self, Self::Error> {
+        <Self as TryFrom<[char; 2]>>::try_from([value[0] as char, value[1] as char])
+            .map_err(|_| value)
+    }
+}
+
+impl FromStr for ChannelId {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.len() != 2 {
+            return Err(format!(
+                "ChannelId must be exactly 2 characters, got {}",
+                s.len()
+            ));
+        }
+        let mut chars = s.bytes();
+        let [Some(ch1), Some(ch2), None] = [chars.next(), chars.next(), chars.next()] else {
+            return Err(format!("Failed to parse channel characters from '{}'", s));
+        };
+        Self::try_from([ch1, ch2]).map_err(|_| format!("Invalid channel characters: '{}'", s))
+    }
+}
```

Comment:
Please avoid to specify error message strings as the error type. Instead, you should add an error type like this:
```rs
#[derive(Debug, Clone, PartialEq, Eq, Hash, Error)]
pub enum ChannelIdParseError {
    #[error("channel id must be exactly 2 ascii characters, got `{0}`")]
    ExpectedTwoAsciiChars(String),
    #[error("channel id must be an alpha numeric to parse as base 62, got {0}")]
    InvalidAsBase62(String),
}
```